examples/LED/LED.ino

```diff
@@ -46,40 +52,80 @@ void setup(void) {
   Serial.println("");
 
   // Wait for connection
-  bool blink = true;
   while (WiFi.status() != WL_CONNECTED) {
     delay(500);
     Serial.print(".");
-    digitalWrite(ledPin, blink ? LOW : HIGH); // active low led
-    blink = !blink;
   }
-  digitalWrite(ledPin, HIGH); // active low led
 
   Serial.println("");
   Serial.print("Connected to ");
   Serial.println(ssid);
   Serial.print("IP address: ");
   Serial.println(WiFi.localIP());
-  adapter = new WebThingAdapter("w25", WiFi.localIP());
+  adapter = new WebThingAdapter("led-lamp", WiFi.localIP());
+
+  lamp.description = "A web connected lamp";
 
-  led.addProperty(&ledOn);
-  adapter->addDevice(&led);
+  lampOn.title = "On/Off";
+  lamp.addProperty(&lampOn);
+
+  toggleInputObj["type"] = "object";
+    JsonObject stateInputProperties =
+      toggleInputObj.createNestedObject("properties");
+  JsonObject stateInput =
+      stateInputProperties.createNestedObject("state");
+  stateInput["type"] = "boolean";
+  lamp.addAction(&toggle);
+
+  adapter->addDevice(&lamp);
   adapter->begin();
+
   Serial.println("HTTP server started");
   Serial.print("http://");
   Serial.print(WiFi.localIP());
   Serial.print("/things/");
-  Serial.println(led.id);
+  Serial.println(lamp.id);
+
+  // set initial values
+  ThingPropertyValue initialOn = {.boolean = true};
+  lampOn.setValue(initialOn);
+  (void)lampOn.changedValueOrNull();
+
 }
 
 void loop(void) {
   adapter->update();
-  bool on = ledOn.getValue().boolean;
-  digitalWrite(ledPin, on ? LOW : HIGH); // active low led
-  if (on != lastOn) {
-    Serial.print(led.id);
-    Serial.print(": ");
-    Serial.println(on);
+  bool on = lampOn.getValue().boolean;
+  if (on) {
+    digitalWrite(lampPin, HIGH);
+  } else {
+    digitalWrite(lampPin, LOW);
   }
-  lastOn = on;
+
+  if (lastOn != on) {
+    lastOn = on;
+  }
+}
+
+void do_toggle(const JsonVariant &input) {
+  Serial.println("toggle call");
+  
+  JsonObject inputObj = input.as<JsonObject>();
+  bool state = inputObj["state"];
+
+  Serial.print("state: ");
+  Serial.println(state);
+
+  if (state) {
```

Comment:
isn't this code redundant? the pin will be written soon after the action handler execution in the loop function.